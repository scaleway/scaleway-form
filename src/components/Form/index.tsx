import type { Decorator } from 'final-form'
import arrayMutators from 'final-form-arrays'
import createDecorator from 'final-form-focus'
import type { ReactNode } from 'react'
import type {
  FormRenderProps,
  FormProps as ReactFinalFormProps,
} from 'react-final-form'
import { Form as ReactFinalForm } from 'react-final-form'
import { ErrorProvider } from '../../providers'
import type { FormErrors } from '../../types'

const focusOnErrors = createDecorator()

export type FormProps<FormValues = unknown> = {
  children?:
    | ((props: FormRenderProps<FormValues, Partial<FormValues>>) => ReactNode)
    | ReactNode
  errors: FormErrors
  /**
   * onRawSubmit is the base onSubmit from final-form
   */
  onRawSubmit: ReactFinalFormProps<FormValues, Partial<FormValues>>['onSubmit']
  initialValues?: Partial<FormValues>
  validateOnBlur?: ReactFinalFormProps<
    FormValues,
    Partial<FormValues>
  >['validateOnBlur']
  validate?: ReactFinalFormProps<FormValues, Partial<FormValues>>['validate']
  /**
   * The form name attribute
   */
  name?: string
  render?: ReactFinalFormProps<FormValues, Partial<FormValues>>['render']
  mutators?: ReactFinalFormProps<FormValues, Partial<FormValues>>['mutators']
  keepDirtyOnReinitialize?: boolean
}

export const Form = <FormValues,>({
  children,
  onRawSubmit,
  errors,
  initialValues,
  validateOnBlur,
  validate,
  name,
  render,
  mutators,
  keepDirtyOnReinitialize,
}: FormProps<FormValues>): JSX.Element => (
  <ReactFinalForm
    initialValues={initialValues}
    validateOnBlur={validateOnBlur}
    validate={validate}
    decorators={[
      focusOnErrors as unknown as Decorator<FormValues, Partial<FormValues>>,
    ]}
    mutators={{
      ...arrayMutators,
      ...mutators,
    }}
    onSubmit={onRawSubmit}
    render={
      render ??
      (renderProps => (
        <ErrorProvider errors={errors}>
          <form noValidate name={name} onSubmit={renderProps.handleSubmit}>
            {typeof children === 'function' ? children(renderProps) : children}
          </form>
        </ErrorProvider>
      ))
    }
    keepDirtyOnReinitialize={keepDirtyOnReinitialize}
  />
)
