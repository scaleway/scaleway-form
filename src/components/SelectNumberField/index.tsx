import { SelectNumber } from '@scaleway/ui'
import type { ComponentProps, FocusEvent, FocusEventHandler } from 'react'
import { useFormField } from '../../hooks'
import type { BaseFieldProps } from '../../types'

type SelectNumberValue = NonNullable<
  ComponentProps<typeof SelectNumber>['value']
>

type SelectNumberValueFieldProps<
  T = SelectNumberValue,
  K = string,
> = BaseFieldProps<T, K> &
  Partial<
    Pick<
      ComponentProps<typeof SelectNumber>,
      | 'disabled'
      | 'maxValue'
      | 'minValue'
      | 'onMaxCrossed'
      | 'onMinCrossed'
      | 'size'
      | 'step'
      | 'text'
      | 'value'
      | 'onChange'
      | 'className'
    >
  > & {
    name: string
    required?: boolean
    onBlur?: FocusEventHandler<HTMLInputElement>
    onFocus?: FocusEventHandler<HTMLInputElement>
  }

export const SelectNumberField = ({
  disabled,
  maxValue,
  minValue,
  name,
  onChange,
  onBlur,
  onFocus,
  onMaxCrossed,
  onMinCrossed,
  required,
  size,
  step,
  text,
  validate,
  value,
  className,
}: SelectNumberValueFieldProps) => {
  const { input } = useFormField(name, {
    disabled,
    required,
    type: 'number',
    validate,
    value,
  })

  return (
    <SelectNumber
      name={name}
      disabled={disabled}
      onBlur={(event: FocusEvent<HTMLInputElement>) => {
        input.onBlur(event)
        onBlur?.(event)
      }}
      onChange={event => {
        input.onChange(event)
        onChange?.(event as number)
      }}
      onFocus={(event: FocusEvent<HTMLInputElement>) => {
        input.onFocus(event)
        onFocus?.(event)
      }}
      maxValue={maxValue}
      minValue={minValue}
      onMinCrossed={onMinCrossed}
      onMaxCrossed={onMaxCrossed}
      size={size}
      step={step}
      text={text}
      value={input.value}
      className={className}
    />
  )
}
