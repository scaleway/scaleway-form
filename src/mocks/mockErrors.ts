import type { FormErrors } from '../types'

export const emailRegex = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i

export const mockErrors: FormErrors = {
  MAX_DATE: ({ maxDate }) => `Date must be lower than ${maxDate?.toString()}`,
  MAX_LENGTH: ({ maxLength }) =>
    `This field should have a length lower than ${maxLength}`,
  MIN_DATE: ({ minDate }) => `Date must be greater than ${minDate?.toString()}`,
  MIN_LENGTH: ({ minLength }) =>
    `This field should have a length greater than ${minLength}`,
  REGEX: ({ regex }) => {
    if (
      (Array.isArray(regex[0])
        ? regex?.[0]?.[0]?.source
        : regex?.[0]?.source) === emailRegex.source
    ) {
      return 'Invalid email'
    }

    return `This field should match the regex ${regex
      .map(r =>
        Array.isArray(r)
          ? r.map(nestedRegex => nestedRegex.source).join(' or ')
          : r.source,
      )
      .join(' and ')}`
  },
  REQUIRED: 'This field is required',
  TOO_HIGH: ({ max }) => `This field is too high (maximum is : ${max})`,
  TOO_LOW: 'This field is too low',
}
