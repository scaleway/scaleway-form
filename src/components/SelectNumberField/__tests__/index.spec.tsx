import { waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import SelectNumberField from '..'
import {
  shouldMatchEmotionSnapshot,
  shouldMatchEmotionSnapshotFormWrapper,
} from '../../../helpers/jestHelpers'
import mockErrors from '../../../mocks/mockErrors'
import Form from '../../Form'

describe('SelectNumberField', () => {
  test('should render correctly', () =>
    shouldMatchEmotionSnapshotFormWrapper(
      <SelectNumberField name="test" value={0} />,
    ))

  test('should render correctly disabled', () =>
    shouldMatchEmotionSnapshotFormWrapper(
      <SelectNumberField name="test" value={10} disabled />,
      {
        transform: ({ getByLabelText }) => {
          const input = getByLabelText('Input')
          expect(input).toBeDisabled()

          const inputMinus = getByLabelText('Minus')
          expect(inputMinus).toBeDisabled()

          const inputPlus = getByLabelText('Plus')
          expect(inputPlus).toBeDisabled()
        },
      },
    ))

  test.only('should trigger events correctly', () => {
    const onFocus = jest.fn(() => {})
    const onChange = jest.fn(() => {})
    const onBlur = jest.fn(() => {})

    return shouldMatchEmotionSnapshotFormWrapper(
      <SelectNumberField
        name="test"
        value={10}
        onChange={onChange}
        onFocus={onFocus}
        onBlur={onBlur}
      />,
      {
        transform: ({ getByLabelText }) => {
          const input = getByLabelText('Input')
          input.focus()
          expect(onFocus).toBeCalledTimes(1)
          // clicking inside field should not trigger any change
          input.click()
          expect(onChange).toBeCalledTimes(0)
          input.blur()
          expect(onBlur).toBeCalledTimes(1)
        },
      },
    )
  })

  test('should trigger event onMinCrossed & onMaxCrossed', () => {
    const onMinCrossed = jest.fn(() => {})
    const onMaxCrossed = jest.fn(() => {})
    const minValue = 5
    const maxValue = 20

    return shouldMatchEmotionSnapshot(
      <Form errors={mockErrors}>
        <SelectNumberField
          maxValue={maxValue}
          minValue={minValue}
          name="test"
          onMinCrossed={onMinCrossed}
          onMaxCrossed={onMaxCrossed}
          value={10}
        />
      </Form>,
      {
        transform: async ({ getByLabelText }) => {
          const input = getByLabelText('Input') as HTMLTextAreaElement
          if (input.parentElement) await userEvent.click(input.parentElement)

          // trigger onMinCrossed
          await userEvent.clear(input)
          await userEvent.type(input, '1')
          await waitFor(() => expect(input.value).toBe('1'))
          input.blur()
          await waitFor(() => expect(input.value).toBe('5'))
          expect(onMinCrossed).toBeCalledTimes(1)

          // trigger onMaxCrossed
          await userEvent.clear(input)
          await userEvent.type(input, '100')
          await waitFor(() => expect(input.value).toBe('100'))
          input.blur()
          await waitFor(() => expect(input.value).toBe('20'))
          expect(onMinCrossed).toBeCalledTimes(1)
        },
      },
    )
  })
})
