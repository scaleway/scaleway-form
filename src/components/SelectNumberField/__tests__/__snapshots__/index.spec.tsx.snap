// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectNumberField should render correctly 1`] = `
<DocumentFragment>
  .cache-8ke8uo {
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  font-weight: 500;
  height: 48px;
  border: 1px solid #d4dae7;
  border-radius: 4px;
}

.cache-bkq66r {
  cursor: pointer;
  background: none;
  border: 0;
  border-radius: 4px;
  color: #c4c9d6;
  padding: 0;
  margin: 0 8px;
}

.cache-bkq66r:hover:not([disabled]) {
  background: #eeeeff;
}

.cache-1ok07vq-sizeStyles {
  vertical-align: middle;
  fill: currentColor;
  height: 26px;
  width: 26px;
  min-width: 26px;
  min-height: 26px;
}

.cache-zw5304 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid transparent;
  max-width: 100%;
}

.cache-zw5304:hover:not([disabled], :focus) {
  border: 1px solid #4f0599;
}

.cache-zw5304:focus-within:not([disabled]) {
  box-shadow: 0px 0px 0px 3px #4F059940;
  border: 1px solid #4f0599;
}

.cache-1afrum9 {
  color: #4a4f62;
  background-color: transparent;
  font-size: 16;
  border: none;
  outline: none;
  position: relative;
  margin-right: 4;
  max-width: 100%;
  font-weight: 500;
  text-align: center;
  -moz-appearance: textfield;
}

.cache-1afrum9::-webkit-outer-spin-button,
.cache-1afrum9::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.cache-1mh6k42 {
  cursor: pointer;
  background: none;
  border: 0;
  border-radius: 4px;
  color: #4f0599;
  padding: 0;
  margin: 0 8px;
}

.cache-1mh6k42:hover:not([disabled]) {
  background: #eeeeff;
}

<form
    novalidate=""
  >
    <div
      class="cache-8ke8uo ermgxlz0"
    >
      <button
        aria-label="Minus"
        class="cache-bkq66r ermgxlz4"
        disabled=""
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H5V11H19V13Z"
          />
        </svg>
      </button>
      <div
        aria-live="assertive"
        class="cache-zw5304 ermgxlz3"
        role="status"
      >
        <input
          aria-label="Input"
          class="cache-1afrum9 ermgxlz2"
          name="test"
          style="width: 25px;"
          type="number"
          value="0"
        />
      </div>
      <button
        aria-label="Plus"
        class="cache-1mh6k42 ermgxlz4"
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          />
        </svg>
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SelectNumberField should render correctly disabled 1`] = `
<DocumentFragment>
  .cache-186d4cs-StyledContainer {
  background-color: #f6f6f8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  font-weight: 500;
  height: 48px;
  border: 1px solid #d4dae7;
  border-radius: 4px;
}

.cache-186d4cs-StyledContainer>.ermgxlz4,
.cache-186d4cs-StyledContainer .ermgxlz2,
.cache-186d4cs-StyledContainer .ermgxlz3 {
  background-color: #f6f6f8;
  border: none;
  color: #c4c9d6;
  opacity: 1;
  cursor: not-allowed;
}

.cache-bkq66r {
  cursor: pointer;
  background: none;
  border: 0;
  border-radius: 4px;
  color: #c4c9d6;
  padding: 0;
  margin: 0 8px;
}

.cache-bkq66r:hover:not([disabled]) {
  background: #eeeeff;
}

.cache-1ok07vq-sizeStyles {
  vertical-align: middle;
  fill: currentColor;
  height: 26px;
  width: 26px;
  min-width: 26px;
  min-height: 26px;
}

.cache-zw5304 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid transparent;
  max-width: 100%;
}

.cache-zw5304:hover:not([disabled], :focus) {
  border: 1px solid #4f0599;
}

.cache-zw5304:focus-within:not([disabled]) {
  box-shadow: 0px 0px 0px 3px #4F059940;
  border: 1px solid #4f0599;
}

.cache-1afrum9 {
  color: #4a4f62;
  background-color: transparent;
  font-size: 16;
  border: none;
  outline: none;
  position: relative;
  margin-right: 4;
  max-width: 100%;
  font-weight: 500;
  text-align: center;
  -moz-appearance: textfield;
}

.cache-1afrum9::-webkit-outer-spin-button,
.cache-1afrum9::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

<form
    novalidate=""
  >
    <div
      class="cache-186d4cs-StyledContainer ermgxlz0"
      disabled=""
    >
      <button
        aria-label="Minus"
        class="cache-bkq66r ermgxlz4"
        disabled=""
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H5V11H19V13Z"
          />
        </svg>
      </button>
      <div
        aria-live="assertive"
        class="cache-zw5304 ermgxlz3"
        role="status"
      >
        <input
          aria-label="Input"
          class="cache-1afrum9 ermgxlz2"
          disabled=""
          name="test"
          style="width: 25px;"
          type="number"
          value="0"
        />
      </div>
      <button
        aria-label="Plus"
        class="cache-bkq66r ermgxlz4"
        disabled=""
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          />
        </svg>
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SelectNumberField should trigger event onMinCrossed & onMaxCrossed 1`] = `
<DocumentFragment>
  .cache-8ke8uo {
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  font-weight: 500;
  height: 48px;
  border: 1px solid #d4dae7;
  border-radius: 4px;
}

.cache-1mh6k42 {
  cursor: pointer;
  background: none;
  border: 0;
  border-radius: 4px;
  color: #4f0599;
  padding: 0;
  margin: 0 8px;
}

.cache-1mh6k42:hover:not([disabled]) {
  background: #eeeeff;
}

.cache-1ok07vq-sizeStyles {
  vertical-align: middle;
  fill: currentColor;
  height: 26px;
  width: 26px;
  min-width: 26px;
  min-height: 26px;
}

.cache-zw5304 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid transparent;
  max-width: 100%;
}

.cache-zw5304:hover:not([disabled], :focus) {
  border: 1px solid #4f0599;
}

.cache-zw5304:focus-within:not([disabled]) {
  box-shadow: 0px 0px 0px 3px #4F059940;
  border: 1px solid #4f0599;
}

.cache-1afrum9 {
  color: #4a4f62;
  background-color: transparent;
  font-size: 16;
  border: none;
  outline: none;
  position: relative;
  margin-right: 4;
  max-width: 100%;
  font-weight: 500;
  text-align: center;
  -moz-appearance: textfield;
}

.cache-1afrum9::-webkit-outer-spin-button,
.cache-1afrum9::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.cache-bkq66r {
  cursor: pointer;
  background: none;
  border: 0;
  border-radius: 4px;
  color: #c4c9d6;
  padding: 0;
  margin: 0 8px;
}

.cache-bkq66r:hover:not([disabled]) {
  background: #eeeeff;
}

<form
    novalidate=""
  >
    <div
      class="cache-8ke8uo ermgxlz0"
    >
      <button
        aria-label="Minus"
        class="cache-1mh6k42 ermgxlz4"
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H5V11H19V13Z"
          />
        </svg>
      </button>
      <div
        aria-live="assertive"
        class="cache-zw5304 ermgxlz3"
        role="status"
      >
        <input
          aria-label="Input"
          class="cache-1afrum9 ermgxlz2"
          name="test"
          style="width: 35px;"
          type="number"
          value="20"
        />
      </div>
      <button
        aria-label="Plus"
        class="cache-bkq66r ermgxlz4"
        disabled=""
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          />
        </svg>
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SelectNumberField should trigger events correctly 1`] = `
<DocumentFragment>
  .cache-8ke8uo {
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  font-weight: 500;
  height: 48px;
  border: 1px solid #d4dae7;
  border-radius: 4px;
}

.cache-bkq66r {
  cursor: pointer;
  background: none;
  border: 0;
  border-radius: 4px;
  color: #c4c9d6;
  padding: 0;
  margin: 0 8px;
}

.cache-bkq66r:hover:not([disabled]) {
  background: #eeeeff;
}

.cache-1ok07vq-sizeStyles {
  vertical-align: middle;
  fill: currentColor;
  height: 26px;
  width: 26px;
  min-width: 26px;
  min-height: 26px;
}

.cache-zw5304 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid transparent;
  max-width: 100%;
}

.cache-zw5304:hover:not([disabled], :focus) {
  border: 1px solid #4f0599;
}

.cache-zw5304:focus-within:not([disabled]) {
  box-shadow: 0px 0px 0px 3px #4F059940;
  border: 1px solid #4f0599;
}

.cache-1afrum9 {
  color: #4a4f62;
  background-color: transparent;
  font-size: 16;
  border: none;
  outline: none;
  position: relative;
  margin-right: 4;
  max-width: 100%;
  font-weight: 500;
  text-align: center;
  -moz-appearance: textfield;
}

.cache-1afrum9::-webkit-outer-spin-button,
.cache-1afrum9::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.cache-1mh6k42 {
  cursor: pointer;
  background: none;
  border: 0;
  border-radius: 4px;
  color: #4f0599;
  padding: 0;
  margin: 0 8px;
}

.cache-1mh6k42:hover:not([disabled]) {
  background: #eeeeff;
}

<form
    novalidate=""
  >
    <div
      class="cache-8ke8uo ermgxlz0"
    >
      <button
        aria-label="Minus"
        class="cache-bkq66r ermgxlz4"
        disabled=""
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H5V11H19V13Z"
          />
        </svg>
      </button>
      <div
        aria-live="assertive"
        class="cache-zw5304 ermgxlz3"
        role="status"
      >
        <input
          aria-label="Input"
          class="cache-1afrum9 ermgxlz2"
          name="test"
          style="width: 25px;"
          type="number"
          value="0"
        />
      </div>
      <button
        aria-label="Plus"
        class="cache-1mh6k42 ermgxlz4"
        type="button"
      >
        <svg
          class="cache-1ok07vq-sizeStyles etwatq50"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          />
        </svg>
      </button>
    </div>
  </form>
</DocumentFragment>
`;
